packages/seojaewan-react/src/hooks/useTotalPrice/useTotalPrice.test.ts

```diff
@@ -0,0 +1,52 @@
+import useTotalPrice from ".";
+import { act, renderHook } from "@testing-library/react";
+
+describe("useTotalPrice 테스트", () => {
+  it("초기 totalPrice 값인 0이 반환된다.", () => {
+    const { result } = renderHook(() => useTotalPrice());
+
+    expect(result.current.totalPrice).toBe(0);
+  });
+
+  it("increaseTotalPrice()에 전달된 1000만큼 totalPrice 값이 증가한다.", () => {
+    const { result } = renderHook(() => useTotalPrice());
+
+    act(() => result.current.increaseTotalPrice(1000));
+    expect(result.current.totalPrice).toBe(1000);
+  });
+
+  it("increaseTotalPrice()에 숫자값이 아닌 'test' 문자열이 전달되면 totalPrice 값이 증가하지 않고 오류가 발생한다.", () => {
+    const { result } = renderHook(() => useTotalPrice());
+    const prevTotalPrice = result.current.totalPrice;
+
+    expect(() => {
+      act(() => result.current.increaseTotalPrice("test"));
+    }).toThrow("increaseTotalPrice()에 전달된 값은 숫자여야 합니다.");
+    expect(result.current.totalPrice).toBe(prevTotalPrice);
+  });
+
+  it("decreaseTotalPrice()에 전달된 500만큼 totalPrice 값이 감소한다.", () => {
+    const { result } = renderHook(() => useTotalPrice());
+
+    act(() => result.current.decreaseTotalPrice(500));
+    expect(result.current.totalPrice).toBe(-500);
+  });
+
+  it("decreaseTotalPrice()에 숫자값이 아닌 'test' 문자열이 전달되면 totalPrice 값이 감소하지 않고 오류가 발생한다.", () => {
+    const { result } = renderHook(() => useTotalPrice());
+    const prevTotalPrice = result.current.totalPrice;
+
+    expect(() => {
+      act(() => result.current.decreaseTotalPrice("test"));
+    }).toThrow("decreaseTotalPrice()에 전달된 값은 숫자여야 합니다.");
+    expect(result.current.totalPrice).toBe(prevTotalPrice);
+  });
+
+  it("increaseTotalPrice()로 1000 증가한 totalPrice를 resetTotalPrice로 0으로 초기화된다.", () => {
+    const { result } = renderHook(() => useTotalPrice());
+
+    act(() => result.current.increaseTotalPrice(1000));
+    act(() => result.current.resetTotalPrice());
+    expect(result.current.totalPrice).toBe(0);
```

Comment:
테스트하려는 게 resetTotalPrice의 동작이라면 increaseTotalPrice로 값을 증가했든 decreaseTotalPrice로 값을 감소시켰든 상관없이 0으로 초기화 되는 게 동작의 정의라고 생각합니다. 따라서  아래와 같이 바꾸면 어떨까 싶습니다. 당연히 테스트의 설명도 달라져야 할 것입니다.
```suggestion
    // 기존에 변경된 값을...
    act(() => result.current.increaseTotalPrice(1000));
    expect(result.current.totalPrice).toBe(1000);

    // 0으로 초기화 한다.
    act(() => result.current.resetTotalPrice());
    expect(result.current.totalPrice).toBe(0);
```

Reply:
아! 기존 값이 1000으로 변경된 것을 확인해야 resetTotalPrice를 통해 0으로 확인하는게 의미가 있겠네요
전 단순하게 `increaseTotalPrice`함수에 대한 테스트가 진행됐기 때문에 생략 했었는데 좀 더 넓게 봐야겠네요!

감사합니닷!